file_catalog/server.py

```diff
@@ -490,50 +464,37 @@ def initialize(self, **kwargs: Any) -> None:  # type: ignore[override]  # pylint
         # pylint: disable=W0201
         self.files_url = os.path.join(self.base_url, 'files')
 
-    @handler.authenticated  # type: ignore[misc]
-    @handler.catch_error  # type: ignore[misc]
+    @fc_auth(prefix=FC_AUTH_PREFIX, roles=FC_AUTH_ROLES)
     async def get(self, uuid: str) -> None:
         """Handle GET request."""
-        try:
-            db_file = await self.db.get_file({'uuid': uuid})
-
-            if db_file:
-                db_file['_links'] = {
-                    'self': {'href': os.path.join(self.files_url, uuid)},
-                    'parent': {'href': self.files_url},
-                }
+        db_file = await self.db.get_file({'uuid': uuid})
+        if not db_file:
+            self.send_error(404, reason='File uuid not found')
```

Comment:
You've already done a lot, so consider this optional. The `raise HTTPError`-syntax is easier to deal with all around.